room/compiler-processing/src/main/java/androidx/room/compiler/processing/ksp/KspFieldElement.kt

```diff
@@ -69,4 +69,17 @@ internal class KspFieldElement(
         declaration = declaration,
         containing = newContaining
     )
+
+    companion object {
+        fun create(
+            env: KspProcessingEnv,
+            declaration: KSPropertyDeclaration
+        ): KspFieldElement {
+            return KspFieldElement(
+                env = env,
+                declaration = declaration,
+                containing = declaration.requireEnclosingTypeElement(env)
```

Comment:
i wonder what happens with these when the field's enclosing element is a File ?
We may want to check for that and throw an error as unsupported, otherwise, the exception may not be very clear.

Reply:
Good catch, this fails for top level properties and functions in KSP (javac works). I'll add a clearer error message and file a bug for this support. I don't need it personally for our usages but it seems like a good thing to support.

Reply:
I created b/184526463 for this

Reply:
thanks. we should eventually model packages here as well but we'll do it once we have a user :)